orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/helpers/ClientSubjectHelper.java

```diff
@@ -101,7 +102,7 @@ public static String calculatePairwiseIdentifier(
             md.update(sectorHost.getBytes(StandardCharsets.UTF_8));
             md.update(subjectID.getBytes(StandardCharsets.UTF_8));
 
-            byte[] bytes = md.digest(salt);
+            byte[] bytes = Objects.nonNull(salt) ? md.digest(salt) : md.digest();
```

Comment:
I don't love the idea of us changing this - I think it's a bit unclear in what cases this is null. I assume it also will change the failure state if the salt is null for some reason where it shouldn't be. I think it would be fine to basically copy most of the logic into the function below, or even commonise the logic above this point - but this feels like commonising for the sake of commonising.